tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py

```diff
@@ -367,6 +420,33 @@ async def pipeline_immediate():
             span.attributes.get(SpanAttributes.DB_STATEMENT), "SET b 2"
         )
 
+    def test_pipeline_immediate_traced_full_time(self):
+        """Command should be traced for coroutine execution time, not creation time."""
+        coro_created_time = None
+        finish_time = None
+
+        async def pipeline_simple():
+            async with self.redis_client.pipeline(
+                transaction=False
+            ) as pipeline:
+                nonlocal coro_created_time
+                nonlocal finish_time
+                pipeline.set("a", 1)
+
+                # delay coroutine creation from coroutine execution
+                coro = pipeline.immediate_execute_command("SET", "b", 2)
+                coro_created_time = time_ns()
+                await coro
+                finish_time = time_ns()
+
+        async_call(pipeline_simple())
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 1)
+        span = spans[0]
+        self.assertTrue(span.start_time > coro_created_time)
+        self.assertTrue(span.end_time < finish_time)
```

Comment:
Is this assertion enough? Wouldn't `span.end_time` be less than `finish_time` even if this fix isn't applied?

Reply:
Yep this second assertion is kind of redundant in terms of testing the fix, but is useful in general to confirm the span timing isn't doing something weird to extend beyond the expected boundary.

The first assertion that checks the start time is the key one in terms of validating the fix.

To really fully validate that the span fully encompasses the coroutine time would require patching every coroutine to return a timestamp to be captured and comparing that. This would kind of nullify the purpose of these integration tests though as then redis isn't being called. We could instead put these tests into the unit test section?
Additionally, this pattern of time testing doesn't seem to be followed anywhere else in this repository so I am hesitant to add too much testing around this accidental bug